find-closest-facility-from-point/src/main/java/com/esri/arcgismaps/sample/findclosestfacilityfrompoint/components/MapViewModel.kt

```diff
@@ -0,0 +1,214 @@
+
+        val incidentPoint = Incident(mapPoint)
+
+        clearIncident(incidentGraphicsOverlay)
+        addIncident(incidentPoint, incidentGraphicsOverlay)
+        currentJob.value = viewModelScope.launch {
+            findRoute(Incident(mapPoint), incidentGraphicsOverlay)
+        }
+    }
+
+    /**
+     * Facility markers are drawn on the map right when the user opens app
+     *
+     */
+    private fun createFacilitiesAndGraphics() {
+
+        // Add all facility centers on the map
+        val facilityUrl = application.getString(R.string.hospital_symbol_url)
+        val facilitySymbol = PictureMarkerSymbol(facilityUrl)
+        facilitySymbol.height = 30F
+        facilitySymbol.width = 30F
+        for (facility in facilitiesList) {
+            val facilityGraphic = Graphic(
+                geometry = facility.geometry, symbol = facilitySymbol
+            )
+            facilityGraphicsOverlay.graphics.add(facilityGraphic)
+        }
+
+    }
+
+    /**
+     * Clear the incident marker from the map
+     *
+     */
+    private fun clearIncident(
+        incidentGraphicsOverlay: GraphicsOverlay
+    ) {
+        incidentGraphicsOverlay.graphics.clear()
+    }
+
+    /**
+     * Add the incident marker on map
+     *
+     */
+    private fun addIncident(
+        incidentPoint: Incident, incidentGraphicsOverlay: GraphicsOverlay
+    ) {
+
+        // Create a cross symbol for the incident
+        val incidentMarker = SimpleMarkerSymbol(
+            style = SimpleMarkerSymbolStyle.Cross, color = Color.black, size = 20.0F
+        )
+
+        // Get the incident's geometry
+        val incidentStopGeometry = incidentPoint.geometry
+
+        // Add graphic to incident graphic overlays
+        incidentGraphicsOverlay.graphics.add(
+            Graphic(
+                geometry = incidentStopGeometry, symbol = incidentMarker
+            )
+        )
+
+    }
+
+    /**
+     * Find the route between the given incident point the closest facility
+     *
+     */
+    private suspend fun findRoute(
+        incidentPoint: Incident, incidentGraphicsOverlay: GraphicsOverlay
+    ) {
+        val closestFacilityTask = ClosestFacilityTask(
+            url = application.getString(R.string.san_diego_network_service_url)
+        )
+
+        // Create a job to find the route
+        try {
+            val closestFacilityParameters: ClosestFacilityParameters =
+                closestFacilityTask.createDefaultParameters().getOrThrow()
+            closestFacilityParameters.setFacilities(facilitiesList)
+            closestFacilityParameters.setIncidents(listOf(incidentPoint))
+
+
+            // Solve a route using the route parameters created
+            val closestFacilityResult: ClosestFacilityResult =
+                closestFacilityTask.solveClosestFacility(closestFacilityParameters).getOrThrow()
+            val rankedFacilitiesList = closestFacilityResult.getRankedFacilityIndexes(0)
```

Comment:
I think it would be nice to indicate what the `0` represents, it is not usually clear without the IDE hints: 

```suggestion
            val rankedFacilitiesList = closestFacilityResult.getRankedFacilityIndexes(incidentIndex = 0)
```